spyder_kernels/ipdb/kernel.py

```diff
@@ -184,8 +206,26 @@ def get_usage(self):
 
     def get_kernel_help_on(self, info, level=0, none_on_fail=False):
         """Help for Pdb commands."""
-        cmd = info['code'].strip()
-        return self.debugger.do_help(cmd)
+        if none_on_fail:
+            return None
+        else:
+            cmd = info['code'].strip()
+            help_on = None
+            try:
+                try:
+                    getattr(self.debugger, 'help_' + cmd)
+                    help_on = self.debugger.do_help(cmd)
+                except AttributeError:
+                    getattr(self.debugger, 'do_' + cmd)
+                    help_on = self.debugger.do_help(cmd)
+            except AttributeError:
+                response = self.do_inspect(info['code'], info['help_pos'])
+                if 'data' in response:
+                    if 'text/plain' in response['data']:
+                        help_on = response['data']['text/plain']
+                if not help_on:
+                    self.debugger.error('No help for %r' % cmd)
```

Comment:
@dalthviz, the test failure is caused because `error` doesn't exist in Pdb for Python 2.

However, you can add it to our SpyderPdb class so that it works for Python 2 and 3.

This is how it's defined in Python 3:

https://github.com/python/cpython/blob/3.6/Lib/pdb.py#L455